### PR TITLE
test/file: try to fix freezing of test_22566

### PR DESCRIPTION
**Pull request by iblislin**
or any suggestion to test `open_cloexec` in `src/support/ios.c` ?

### CODE REVIEW

**Comment by vtjnash** [comment]
What does this do? A simple blocking read/write of a byte would be preferable to a timed sleep.

**Comment by iblis17** [comment]
@vtjnash recent CI build froze quite frequently after my PR #22566 merged.
I can reproduce it on my local via runing `gmake test` 2~5 times.
I attached my GDB to check the bt of works, one of the workers stucks at the reader's open (`r = open(fn, "r")`).
So I guess that there is race condiction between the writer's open and reader's, but still no idea about the roo cause.

**Comment by iblis17** [comment]
> A simple blocking read/write of a byte would be preferable to a timed sleep.

I will try it tmr. :)

**Comment by iblis17** [comment]
```patch
+++ b/test/file.jl
@@ -1141,11 +1141,12 @@ if !Sys.iswindows()
         fn = tempname()
         run(`mkfifo $fn`)

-        script = "x = open(\"$fn\", \"w\"); sleep(0.1); close(x); quit()"
+        script = "x = open(\"$fn\", \"w\"); write(x, 0x42); flush(x); close(x); quit()"
         cmd = `$(Base.julia_cmd()) --startup-file=no -e $script`
         open(pipeline(cmd, stderr=STDERR))

         r = open(fn, "r")
+        @test read(r, Int64) == 66
         close(r)

         rm(fn)
```
got this
```
file: Error During Test
  Test threw an exception of type EOFError
  Expression: read(r, Int64) == 66
  EOFError: read end of file
  Stacktrace:
   [1] read at ./iostream.jl:202 [inlined]
   [2] test_22566() at /usr/home/iblis/git/julia/test/file.jl:1149
   [3] macro expansion at /usr/home/iblis/git/julia/test/file.jl:1157 [inlined]
   [4] anonymous at ./<missing>:?
   [5] include_relative(::Module, ::String) at ./loading.jl:464
   [6] include at ./sysimg.jl:14 [inlined]
   [7] include(::String) at /usr/home/iblis/git/julia/test/testdefs.jl:11
   [8] macro expansion at /usr/home/iblis/git/julia/test/testdefs.jl:18 [inlined]
   [9] macro expansion at ./test.jl:943 [inlined]
   [10] macro expansion at ./util.jl:379 [inlined]
   [11] macro expansion at /usr/home/iblis/git/julia/test/testdefs.jl:17 [inlined]
   [12] anonymous at ./<missing>:?
```

or I'm okay with getting rid of this test case..

**Comment by vtjnash** [comment]
> write(x, 0x42);

You're only writing a byte, but attempting to read an Int64

File: test/file.jl [hunk 1]
```diff
@@ -1141,7 +1141,7 @@ if !Sys.iswindows()
         fn = tempname()
         run(`mkfifo $fn`)
 
-        script = "x = open(\"$fn\", \"w\"); close(x)"
+        script = "x = open(\"$fn\", \"w\"); sleep(0.1); close(x); quit()"
```
Comment (vtjnash) [inline test/file.jl]: while we're changing this test, can you replace `\"$fn\"` with `$(repr(fn))` for me too? (that'll make it robust against other random meta-characters happening to to get inject into temp path)

Comment (iblis17) [inline test/file.jl]: 🆗

**Comment by iblis17** [comment]
recent CI freeze is this
```julia
    frame #56: 0x0000000800a600b5 libjulia.so.0.7`start_task + 437 at task.c:268 [opt]
(lldb) p jl_lineno
(int) $0 = 1027
(lldb) p jl_filename
(const char *) $1 = 0x000000080d86b0d8 "/home/julia/ci/worker/11rel-amd64/build/test/file.jl"
(lldb) thread list
Process 85568 stopped
* thread #1: tid = 101577, 0x0000000802189498 libc.so.7`__sys_openat + 8
  thread #2: tid = 101161, 0x00000008021892ba libc.so.7`_sigwait + 10
```
that test case has similar structure with the test case disabled in this PR.

**Pull request merged**